tests/e2e/magic_functions/functions/c_write.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "mg_procedure.h"
+
+static void try_to_write(struct mgp_list *args, mgp_func_context *ctx, mgp_func_result *result,
+                         struct mgp_memory *memory) {
+  mgp_value *value{};
+  mgp_vertex *vertex{};
+  mgp_list_at(args, 0, &value);
+  mgp_value_get_vertex(value, &vertex);
+
+  const char *name;
+  mgp_list_at(args, 1, &value);
+  mgp_value_get_string(value, &name);
+
+  mgp_list_at(args, 2, &value);
+
+  auto err_code = mgp_vertex_set_property(vertex, name, value);  // This should set an error
+  if (err_code != MGP_ERROR_NO_ERROR) {
+    mgp_func_result_set_error(result, "Cannot set property in the function!", memory);
+    return;
+  }
+
+  err_code = mgp_func_result_set_value(result, value, memory);
+  if (err_code != MGP_ERROR_NO_ERROR) {
+    mgp_func_result_set_error(result, "Failed to construct return value!", memory);
+    return;
+  }
+}
+
+// Each module needs to define mgp_init_module function.
+// Here you can register multiple procedures your module supports.
+extern "C" int mgp_init_module(struct mgp_module *module, struct mgp_memory *memory) {
+  {
+    mgp_func *func;
+    auto err_code = mgp_module_add_function(module, "try_to_write", try_to_write, &func);
+    if (err_code != MGP_ERROR_NO_ERROR) {
+      return 1;
+    }
+
+    mgp_type *type_vertex{};
+    mgp_type_node(&type_vertex);
+    err_code = mgp_func_add_arg(func, "argument", type_vertex);
+    if (err_code != MGP_ERROR_NO_ERROR) {
+      return 1;
+    }
+
+    mgp_type *type_string{};
+    mgp_type_string(&type_string);
+    err_code = mgp_func_add_arg(func, "name", type_string);
+    if (err_code != MGP_ERROR_NO_ERROR) {
+      return 1;
+    }
+
```

Comment:
plz test `mgp_func_add_opt_arg` as well!

Reply:
Added this to the `c_read.cpp`